js/functions.js

```diff
@@ -16,55 +18,77 @@ function menuShow() {
 }
 
 
-function dropdownShow() {
-    var items = document.getElementsByClassName("navbar-collapse-dropdown-item");
+function dropdownShow()
+{
+    let items = document.getElementsByClassName("navbar-collapse-dropdown-item");
 
     for (i = 0; i < items.length; i++) {
         if (items[i].classList.contains("hide")) {
-            items[i].classList.remove("hide")
-        } else {
-            items[i].classList.add("hide");
+            items[i].classList.remove("hide");
+            continue;
         }
+
+        items[i].classList.add("hide");
     }
 }
 
-function dropdownIsOpen(){
-    var items = document.getElementsByClassName("navbar-collapse-dropdown-item");
+function dropdownIsOpen()
+{
+    let items = document.getElementsByClassName("navbar-collapse-dropdown-item");
     return !items[i].classList.contains("hide") ;
 }
 
-function validateForm(){
+function validateForm()
+{
+    let msg = "", type = "", utilizador = "", dominio="" ;
 
-    var msg = "", nome = "", email = "", type = "", utilizador = "", dominio="" ;
+    let nome  = document.getElementById( "nome" ).value;
+    let email = document.getElementById( "email" ).value;
 
-    var nome  = document.getElementById( "nome" ).value;
-    var email = document.getElementById( "email" ).value;
+    if( nome == "" || email == "" ) {
+        msg = "Nome e Email são obrigatórios!";
+        type = "error-message";
 
-    if( nome != "" && email != "" ){
+        generateFormValidationMessage(msg, type);
 
-        if ( email.indexOf('@') != -1 && email.indexOf('.') != -1 ) {
+        return;
+    }
 
-            utilizador = email.substring( 0, email.indexOf('@') ) ;
-            dominio = email.substring( email.indexOf('@') , email.length ) ;
+    if (validateEmailForRegister(email) !== true) {
+        return;
+    }
 
-            if ( utilizador.length >= 1 && dominio.length >= 3 ){
-                msg = "Sucesso! Sua inscrição foi confirmada!"
-                type = "success-message";
-            } else {
-                msg = "Informe um e-mail correto!"
-                type = "error-message";
-            }
-        }else{
-            msg = "Informe um e-mail correto!"
-            type = "error-message";
-        }
-    } else {
-        msg="Informe os campos corretamente!"
+    msg = "Sucesso! Sua inscrição foi confirmada!";
+    type = "success-message";
+
+    generateFormValidationMessage(msg, type);
+}
+
+function generateFormValidationMessage (msg, type)
+{
+    document.getElementById( 'form-output').innerHTML = "<p class=\""+type+"\"> " + msg + "</>"
+}
+
+function validateEmailForRegister(email)
+{
+    if ( email.indexOf('@') == -1 || email.indexOf('.') == -1 ) {
```

Comment:
This validation is kind of bogus. For example, it would allow `a.fake@domain`. Maybe using http://emailregex.com/ would be better.